src/index.ts

```diff
@@ -0,0 +1,607 @@
+import * as fs from "fs";
+import FormData from "form-data";
+import { spawn, execSync } from "child_process";
+import { ArgumentParser } from "argparse";
+import { ApiScanFormat, FormTypes, OnFailure, ScanMode, SubmitActions } from "./utils/enums";
+import { exit } from "process";
+import SOOSAnalysisApiClient from "@soos-io/api-client/dist/api/SOOSAnalysisApiClient";
+import {
+  getEnvVariable,
+  isUrlAvailable,
+  convertStringToB64,
+  sleep,
+} from "@soos-io/api-client/dist/utilities";
+import {
+  ScanStatus,
+  ScanType,
+  soosLogger,
+  LogLevel,
+  OutputFormat,
+  SOOS_CONSTANTS,
+} from "@soos-io/api-client";
+import { ZAPCommandGenerator, CONSTANTS } from "./utils";
+
+export interface SOOSDASTAnalysisArgs {
+  apiKey: string;
+  apiURL: string;
+  appVersion: string;
+  branchName: string;
+  branchUri: string;
+  buildUri: string;
+  buildVersion: string;
+  clientId: string;
+  commitHash: string;
+  integrationName: string;
+  integrationType: string;
+  logLevel: LogLevel;
+  onFailure: OnFailure;
+  operatingEnvironment: string;
+  projectName: string;
+  scriptVersion: string;
+  targetURL: string;
+  scanMode: ScanMode;
+  debug: boolean;
+  ajaxSpider: boolean;
+  contextFile: string;
+  fullScanMinutes: number;
+  apiScanFormat: ApiScanFormat;
+  authUsername: string;
+  authPassword: string;
+  authLoginURL: string;
+  authUsernameField: string;
+  authPasswordField: string;
+  authSubmitField: string;
+  authSecondSubmitField: string;
+  authSubmitAction: SubmitActions;
+  authFormType: FormTypes;
+  authDelayTime: number;
+  authVerificationURL: string;
+  requestCookies: string;
+  requestHeaders: string;
+  reportRequestHeaders: boolean;
+  bearerToken: string;
+  oauthTokenUrl: string;
+  oauthParameters: string;
+  updateAddons: boolean;
+  disableRules: string;
+  otherOptions: string;
```

Comment:
did we lose rules? or can we do that via otherOptions?

Reply:
great - we'll need to be able to pass a config file like this - https://www.zaproxy.org/faq/how-do-you-find-out-what-key-to-use-to-set-a-config-value-on-the-command-line/
```
(at the bottom)

Reply:
was already able to test it